### PR TITLE
Fix to properly handle invalid SMS URL

### PR DESCRIPTION
**Pull request by dilin993**
## Purpose
When invalid SMS URL is given for SMS OTP configuration log error log is shown. But no feedback is given via the UI.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by dilin993** [comment]
Fix the issue https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/issues/43

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 1]
```diff
@@ -935,7 +935,14 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
         String encodedMobileNo = encoder.encode(mobile);
         smsUrl = smsUrl.replaceAll("\\$ctx.num", encodedMobileNo).replaceAll("\\$ctx.msg",
                 smsMessage.replaceAll("\\s", "+") + otpToken);
-        URL smsProviderUrl = new URL(smsUrl);
+        URL smsProviderUrl = null;
+        try {
+            smsProviderUrl = new URL(smsUrl);
+        } catch (MalformedURLException e) {
+            log.error("Invalid SMS URL provided : " + e);
```
Comment (omindu) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: `log.error("Error while parsing SMS provider URL: " + smsUrl, e);`

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 2]
```diff
@@ -935,7 +935,14 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
...
+            context.setProperty(SMSOTPConstants.ERROR_CODE, "Invalid SMS URL provided");
```
Comment (ruwanta) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Can we change this to "The SMS URL does not conform to URL specification"

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 1 above)
Comment (dilin993) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Changed the error message as avove

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 2 above)
Comment (dilin993) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: +1 Let's change this

**Pull request merged**